### ISSUE TITLE
Select URL when opening new tab?

### ISSUE THREAD

**Issue by vpzomtrrfrt** [LadybirdBrowser/ladybird#137]
When I open a new tab, I do so because I want to type a URL or search. Currently, this is somewhat annoying in Ladybird because the location field already contains "about:newtab".

Firefox seems to deal with this by leaving the location field blank when opening new tabs, but automatically selecting its contents would also avoid the issue.

This patch seems to work:
```patch
diff --git a/Ladybird/Qt/Tab.cpp b/Ladybird/Qt/Tab.cpp
index d00db16a71..955120c4af 100644
--- a/Ladybird/Qt/Tab.cpp
+++ b/Ladybird/Qt/Tab.cpp
@@ -156,6 +156,9 @@ Tab::Tab(BrowserWindow* window, WebContentOptions const& web_content_options, St
 
     view().on_url_change = [this](auto const& url) {
         m_location_edit->setText(qstring_from_ak_string(url.serialize()));
+
+        if (m_location_edit->hasFocus())
+            m_location_edit->selectAll();
     };
 
     QObject::connect(m_location_edit, &QLineEdit::returnPressed, this, &Tab::location_edit_return_pressed);
```

**Comment by skyrising** [comment]
There's a PR for hiding the URL: #128

**Comment by hughdavenport** [comment]
That does have potential, but doesn't cover all situations.
- Starting ladybird does not select
- A new window does not select
- A new tab with a custom url does select, where firefox doesn't and doesn't focus

I did a more detailed post on that PR for hiding the URL which I think outlines what the current state of other browsers behaviour is and what the best outcome for ladybird is: https://github.com/LadybirdBrowser/ladybird/pull/128#issuecomment-2162042437

TLDR, I think hiding the `about:...` pages would be the best solution, which I think would cover your usecase wanted (and I would like too! It is quite annoying ha)
